.ci/docker/golang-mage/Dockerfile

```
@@ -1,25 +1,11 @@
 ARG GO_VERSION=1.13.8
 FROM golang:${GO_VERSION}
 
-WORKDIR /tools
-
-RUN echo "Building from golang:${GO_VERSION}"
-RUN git clone https://github.com/magefile/mage \
-  && cd mage \
-  && go run bootstrap.go \
-  && mage -version \
-  && cd /tools \
-  && go get golang.org/x/tools/cmd/benchcmp \
-  && go get github.com/t-yuki/gocover-cobertura \
-  && go get github.com/jstemmer/go-junit-report \
-  && go get github.com/kardianos/govendor \
-  && go get github.com/elastic/go-licenser \
-  && go get github.com/pierrre/gotestcover \
-  && go get github.com/stretchr/testify/assert \
-  && go get golang.org/x/tools/cmd/goimports
+ENV TOOLS=/tools
+WORKDIR $TOOLS
+COPY go.mod .
+RUN go mod download
```

mage install is still needed

```suggestion
RUN go mod download
RUN git clone https://github.com/magefile/mage \	RUN go mod download
  && cd mage \	
  && go run bootstrap.go \	
  && mage -version
```

---

We wanted to avoid a global install, instead resolved the mage issues by using the make commands. (Thanks @axw for your help on this).

---

This container is used by the CI, it forces the CI to use fixed tools, do it should install mage. This container is a repeatable environment where we can build the apm-server. You can have Makefile magic to install mage if you want to help people to prepare the environment, but introduce those tools installed on the build process make it difficult to track and introduces errors not related to the build.

---

@kuisathaverat we're using Go modules now, and we have mage as one of the entries in `go.mod`. When we build mage (as part of the Make command), it builds a specific version.

---

I do not see the changes in the makefile here, so I guess that I have to take a look, but from my perspective have a `make` command that calls a `mage` command sounds a little complex, knowing that both are build tools with the same purpose.

---

Please take a look and let me know what you think. We tried to be in line with the go modules changes, but can go back to install mage globally for CI if it's important.

---

The new Makefile is much simpler. What makes me think about why we do not move everything to `mage` and remove the `Makefile`? this makes `mage` a pre-requisite but wi will have only one build definition `magefile.go` and only one build tool `mage`

---

I saw that `go mod download` does not install mage binary

```
root@c3553c27f088:/go/src/github.com/elastic/apm-server# find /go -name mage
/go/pkg/mod/github.com/magefile/mage@v1.9.0/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200312020002-9735c6d51540/dev-tools/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200312020002-9735c6d51540/metricbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200312020002-9735c6d51540/winlogbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200312020002-9735c6d51540/filebeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200312020002-9735c6d51540/heartbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200312020002-9735c6d51540/packetbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200312020002-9735c6d51540/auditbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200312020002-9735c6d51540/x-pack/functionbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200306124118-bbf9d6697f4a/dev-tools/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200306124118-bbf9d6697f4a/metricbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200306124118-bbf9d6697f4a/winlogbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200306124118-bbf9d6697f4a/filebeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200306124118-bbf9d6697f4a/heartbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200306124118-bbf9d6697f4a/packetbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200306124118-bbf9d6697f4a/auditbeat/scripts/mage
/go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20200306124118-bbf9d6697f4a/x-pack/functionbeat/scripts/mage
/go/pkg/mod/github.com/elastic/go-txfile@v0.0.7/dev-tools/lib/mage
/go/pkg/mod/cache/download/github.com/magefile/mage
```

---

There is a make target to build `mage`

```
BIN_MAGE=$(GOOSBUILD)/bin/mage

# BIN_MAGE is the standard "mage" binary.
$(BIN_MAGE): go.mod
	go build -o $@ github.com/magefile/mage
```